### PR TITLE
Full spark integration

### PR DESCRIPTION
**Pull request by elijahbenizzy**
[Short description explaining the high-level reason for the pull request]

## Checklist

- [ ] PR has an informative and human-readable title (this will be pulled into the release notes)
- [ ] Changes are limited to a single goal (no scope creep)
- [ ] Code passed the pre-commit check & code is left cleaner/nicer than when first encountered.
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future TODOs are captured in comments
- [ ] Project documentation has been updated if adding/changing functionality.

### CODE REVIEW

**Comment by elijahbenizzy** [comment]
Things to look over:
- [ ] Documentation -- where else can we add references?
- [ ] API -- we'll be locked in after this
- [ ] Clarity of explanations -- I can do a better job here I think

File: hamilton/experimental/h_spark.py [hunk 1]
```diff
@@ -206,46 +215,41 @@ def python_to_spark_type(python_type: Type[Union[int, float, bool, str, bytes]])
     _list = (list[int], list[float], list[bool], list[str], list[bytes])
 
 
-def get_spark_type(
-    actual_kwargs: dict, df: DataFrame, hamilton_udf: Callable, return_type: Any
-) -> types.DataType:
+def get_spark_type(return_type: Any) -> types.DataType:
     if return_type in (int, float, bool, str, bytes):
         return python_to_spark_type(return_type)
+    elif return_type in (list[int], list[float], list[bool], list[str], list[bytes]):
```
Comment (skrawcz) [inline hamilton/experimental/h_spark.py:221]: this should reference `_list` above.

File: hamilton/experimental/h_spark.py [hunk 2]
```diff
@@ -259,24 +263,22 @@ def _bind_parameters_to_callable(
...
+    bind_parameters = {}
+    for input_name, (type_, dep_type) in node_input_types.items():
         if input_name in df_columns:
             params_from_df[input_name] = column(input_name)
         elif input_name in actual_kwargs and not isinstance(actual_kwargs[input_name], DataFrame):
-            hamilton_udf = functools.partial(
-                hamilton_udf, **{input_name: actual_kwargs[input_name]}
-            )
-        elif node_input_types[input_name][1] == DependencyType.OPTIONAL:
-            pass
```
Comment (skrawcz) [inline hamilton/experimental/h_spark.py:273]: where did this change go?

Comment (elijahbenizzy) [reply hamilton/experimental/h_spark.py:273]: See the next line

Comment (skrawcz) [reply hamilton/experimental/h_spark.py:273]: oh you check for required, else skip.

File: hamilton/experimental/h_spark.py (hunk 1 above)
Comment (skrawcz) [reply hamilton/experimental/h_spark.py:221]: else this will break for python < 3.9

Comment (elijahbenizzy) [reply hamilton/experimental/h_spark.py:221]: Yeah this was lost in the merge

**Review by skrawcz — APPROVED** [review]

**Pull request merged**